heron/common/src/cpp/network/baseserver.cpp

```diff
@@ -106,6 +106,17 @@ sp_int32 BaseServer::Start_Base() {
     return -1;
   }
 
+  // fetch the port after bind/listen port 0
+  if (AF_INET == options_.get_sin_family() && 0 == options_.get_port()) {
+    if (getsockname(listen_fd_, serv_addr, &sockaddr_len) != 0)  {
```

Comment:
@huijunwu - I saw this fragment of code repeating in another PR. Can you write a small utility function in sockutils.cpp - getPort - given the socket and use that function here?